src/dock-api-retry.js

```diff
@@ -0,0 +1,266 @@
+  maxAttempts: 2,
+  /** Amount of blocks to be fetched in addition to the strict amount of blocks. It will cover timing for fetching the block data. */
+  fetchGapBlocks: 1,
+};
+
+export const FASTBLOCK_CONFIG = {
+  /** Amount of blocks to wait for the extrinsic to be finalized before retrying. */
+  finalizedTimeoutBlocks: 8,
+  /** Amount of blocks to wait for the extrinsic to be included in the block before retrying. */
+  inBlockTimeoutBlocks: 4,
+  /** Amount of blocks to wait before a retry attempt. */
+  retryDelayBlocks: 6,
+  /** Block time in ms. */
+  blockTimeMs: FASTBLOCK_TIME_MS,
+  /** Max retry attempts (doesn't include initial request). */
+  maxAttempts: 3,
+  /** Amount of blocks to be fetched in addition to the strict amount of blocks. It will cover timing for fetching the block data. */
+  fetchGapBlocks: 5,
+};
+
+/**
+ * Properties that won't be patched/visited during the patching.
+ */
+const BlacklistedProperties = new Set([
+  'meta',
+  'registry',
+  'toJSON',
+  'is',
+  'creator',
+  'hash',
+  'key',
+  'keyPrefix',
+]);
+
+/**
+ * Recursively patches supplied object property and all underlying objects, so all functions will attempt to retry 2 times and
+ * will throw an error if there's no result within the `8 seconds` timeout.
+ *
+ * @param {*} obj
+ * @param {*} prop
+ * @param {string[]} [path=[]]
+ */
+const wrapFnWithRetries = (obj, prop, path = []) => {
+  const value = obj[prop];
+  if (
+    BlacklistedProperties.has(prop)
+    || !value
+    || (typeof value !== 'object' && typeof value !== 'function')
+  ) {
+    return;
+  }
+
+  try {
+    let newValue;
+    if (typeof value !== 'function') {
+      newValue = Object.create(Object.getPrototypeOf(value));
+    } else {
+      newValue = async function with8SecsTimeoutAnd2Retries(...args) {
+        const wrappedFn = () => value.apply(this, args);
+        wrappedFn.toString = () => value.toString();
+
+        return await retry(wrappedFn, 8e3, {
+          maxAttempts: 2,
+          delay: 5e2,
+          onTimeoutExceeded: (retrySym) => {
+            console.error(`\`${path.concat('.')}\` exceeded timeout`);
+
+            return retrySym;
+          },
+        });
+      };
+
+      Object.setPrototypeOf(newValue, Object.getPrototypeOf(value));
+    }
+
+    for (const key of Object.keys(value)) {
+      newValue[key] = value[key];
+      wrapFnWithRetries(newValue, key, path.concat(key));
+    }
+
+    // eslint-disable-next-line no-param-reassign
+    delete obj[prop];
+    Object.defineProperty(obj, prop, {
+      value: newValue,
+    });
+  } catch (err) {
+    console.error(
+      `Failed to wrap the prop \`${prop}\` of \`${obj}\`: \`${
+        err.message || err
+      }\``,
+    );
+  }
+};
+
+/**
+ * Patches the query API methods, so they will throw an error if there's no result within the `8 seconds` timeout after `2` retries.
+ *
+ * @param {*} queryApi
+ */
+export const patchQueryApi = (queryApi) => {
```

Comment:
It will wrap all methods with retry functions to call the underlying method again if time is exceeded. It might fail because of the way misc JS platform will handle property overrides.

Reply:
> It might fail because of the way misc JS platforms will handle property overrides.

Can you elaborate? 

Have any tests failed because of this?